### PR TITLE
HeatmapNG: explicitly pass scales for cursor sync config

### PR DESCRIPTION
**Pull request by leeoniya**
temporary work-around for https://github.com/leeoniya/uPlot/issues/715

was causing sync glitches and console errors on mousemoves.

(also a bunch of typings fixes)

### CODE REVIEW

File: public/app/plugins/panel/heatmap/utils.ts [hunk 1]
```diff
@@ -570,6 +570,7 @@ export function prepConfig(opts: PrepConfigOpts) {
   if (sync && sync() !== DashboardCursorSync.Off) {
     cursor.sync = {
       key: '__global_',
+      scales: [xScaleKey, yScaleKey],
```
Comment (leeoniya) [inline public/app/plugins/panel/heatmap/utils.ts:573]: ^^ in case you missed it :D

File: public/app/plugins/panel/heatmap/utils.ts [hunk 2]
```diff
@@ -257,6 +257,7 @@ export function prepConfig(opts: PrepConfigOpts) {
     return builder; // early abort (avoids error)
   }
 
+  // eslint-ignore @typescript-eslint/no-explicit-any
```
Comment (leeoniya) [inline public/app/plugins/panel/heatmap/utils.ts:260]: @ashharrison90 @joshhunt any advice on how to avoid this one?

Comment (joshhunt) [reply public/app/plugins/panel/heatmap/utils.ts:260]: Yup - you're running into the big type holes of grafana. I would say leave this in at the moment (It's a good marker for something that we would want to address later), and we can come up with a way to try and type field configs on data frames.

**Review by ryantxu — APPROVED** [review]

File: public/app/plugins/panel/heatmap/utils.ts (hunk 2 above)
Comment (joshhunt) [reply public/app/plugins/panel/heatmap/utils.ts:260]: My first pass at this would be to give DataFrame a generic to type the Field custom properties.

**Pull request merged**